pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -413,6 +413,12 @@ def _check_device_config_and_set_final_flags(
         self._num_nodes_flag = int(num_nodes) if num_nodes is not None else 1
         self._devices_flag = devices
 
+        if self._devices_flag in ([], 0, "0"):
+            raise MisconfigurationException(
+                f"`Trainer(devices={self._devices_flag!r})` value is not a valid input"
+                f" using {self._accelerator_flag} accelerator."
+            )
```

Comment:
if someone pass in `Trainer(accelerator=CustomAccelerator(...))` I think accelerator_flag will points to that class instance. Should we put `self.accelerator.__class__.__name__`?

Reply:
Updated!